### ISSUE TITLE
Provide previews of Readmes on search result pages

### ISSUE THREAD

**Issue by jayoung-lee** [dart-lang/pub-dev#4367]
In a recent user study, we observed that users open the package page and close the tab immediately, when there is no clear Readme. If we provide a preview of Readme on the search result page, we can help users avoid visiting packages with low-quality Readme and save time for the searchers. Literature also suggests to use thumbnail representations of documents for rapid assessment of document relevance prior to selection. (Google Search removed it due to low usage, but our users may benefit by just seeing what headings or images a Readme page has.)

**Comment by elbeicktalat** [comment]
Nice idea but how it will be implemented? 
Also which piece of the readme you'll show?
Also request this data for the entire search results will be expensive in terms of time loading, what will happen if we make the user loses time instead of saving time?

Thanks!

**Comment by jayoung-lee** [comment]
@elbeicktalat  This is in the stage of ideation, so there's no details for implementation yet :)

**Comment by elbeicktalat** [comment]
@jayoung-lee I understand that, I just comment on this maybe it can help some one in the team. I'm not saying this idea is bad or something else no, it's just simple questions we should look.
Thanks!

**Comment by jayoung-lee** [comment]
@elbeicktalat Yeah, thanks for leaving a comment. They're great material for deep thought!
